### DISCUSSION TITLE
Help with understanding detection

### DISCUSSION THREAD

**Discussion by bodgerbob** [blakeblackshear/frigate#6642] [Ask A Question]
Hi, I am hoping that someone can help me understand why my motion mask and detection masks are not working as I expect.

The location of my one of my cameras is such that it points fairly directly onto the windscreen of my car. When it is sunny the reflection of clouds triggers motion, and often the reflection is detected as a person.
I have created both detection and motion masks of the windscreen and this appeared to solve the problem, however it has started detecting person objects again, without me making any other changes.
Can someone advise me as to what I am doing wrong please?

I have attached images of the mask and a typical snapshot. An extract of my config is below:

            
    motion:
      mask:
        - 0,0,0,153,236,69,238,43,640,57,640,0
        - 48,114,232,82,283,172,32,222    # windscreen
      threshold: 10   #  1 to 255. 
      
      

    detect:
      width: 640
      height: 480
      fps: 5
      enabled: True
      
    objects:
      filters:
        person:
          # people can't be in this area
          mask:
            - 0,0,0,153,236,69,238,43,640,57,640,0
            - 48,114,232,82,283,172,32,222    # windscreen

Thank you

![masks](https://github.com/blakeblackshear/frigate/assets/79364062/955f5899-a36a-4bdc-94a0-d3dffbf8ca2a)
![snapshot](https://github.com/blakeblackshear/frigate/assets/79364062/2788d5d8-0c06-4271-9c86-229c006f6f65)

**Comment by NickM-27** [comment]
Motion masks will not stop objects from being detected. Object masks stop objects from being detected based on if the bottom center pixel of the detected objects bounding box is within the mask. In this case the "person" bottom center pixel is outside of the object mask so it is not being masked out.

**Comment by bodgerbob** [comment]
Thank you for the explanation. I can now make changes accordingly.
